geomstats/visualization/symmetric_positive_definite.py

```diff
@@ -0,0 +1,598 @@
+
+    This class provides all the essential methods for the
+    visualization of the manifold of the Symmetric Positive
+    Definite matrices.
+
+    Parameters
+    ----------
+    max_z: int
+    The scaling factor of the manifold
+
+    Attributes
+    ----------
+    currZ: int
+    The scaling factor of the manifold
+
+    spdPointViz:
+    Class used to plot points on the manifold SPD(2).
+    Elements S of the manifold of 2D Symmetric Positive
+    Definite matrices can be conveniently represented by ellipses.
+
+    spdManifold:
+    Class for the manifold of symmetric positive definite (SPD) matrices.
+    Takes as input n (int) Integer representing the shape of the matrices: nxn
+    By default n is set to 2 to in order for the visualization to be feasible.
+
+    metric:
+    Compute the affine-invariant exponential map.Compute the Riemannian
+    exponential at point base_point of tangent vector tangent_vec wrt
+    the metric defined in inner_product.
+    This gives a symmetric positive definite matrix.
+
+    References
+    ----------
+    [1] Miolane, Nina, et al. "Geomstats: a Python package for
+    Riemannian geometry in machine learning."
+    Journal of Machine Learning Research 21.223 (2020): 1-9.
+    """
+
+    def __init__(self, max_z=1):
+        """Initialize class parameters of the manifold."""
+        self.max_z = float(max_z)
+        self.currZ = self.max_z
+        self.ax = None
+        self.spdPointViz = Ellipses()
+        self.spdManifold = SPDMatrices(2)
+        self.metric = SPDAffineMetric(2)
+
+    def cuboid_data(self, o, size=(1, 1, 1)):
+        """Generate cuboid data for plotting."""
+        X = [
+            [[0, 1, 0], [0, 0, 0], [1, 0, 0], [1, 1, 0]],
+            [[0, 0, 0], [0, 0, 1], [1, 0, 1], [1, 0, 0]],
+            [[1, 0, 1], [1, 0, 0], [1, 1, 0], [1, 1, 1]],
+            [[0, 0, 1], [0, 0, 0], [0, 1, 0], [0, 1, 1]],
+            [[0, 1, 0], [0, 1, 1], [1, 1, 1], [1, 1, 0]],
+            [[0, 1, 1], [0, 0, 1], [1, 0, 1], [1, 1, 1]],
+        ]
+        X = np.array(X).astype(float)
+        X -= 0.5
+        for i in range(3):
+            X[:, :, i] *= size[i]
+        X += np.array(o)
+        return X
+
+    def plot_cube_at(self, positions, sizes=None, colors=None, **kwargs):
+        """Plot cube relative to specific coordinates.
+
+        Parameters
+        ----------
+        positions : array-like, size [1,3]
+            Cordinates of the specific point
+        sizes : list of tuples
+            Size of the cube-shaped tangent space
+        colors: string, optional (default=None)
+            Specifies the color of the cube
+
+        Returns
+        -------
+            Figure plot
+        """
+        if not isinstance(colors, (list, np.ndarray)):
+            colors = ["C0"] * len(positions)
+        if not isinstance(sizes, (list, np.ndarray)):
+            sizes = [(1, 1, 1)] * len(positions)
+        g = []
```

Comment:
avoid single letter variable names bcs it is not readable --> rename g into something more explicit